### PR TITLE
Proper error message for missing variables in inline assembly

### PR DESCRIPTION
**Pull request by axic**
Part of #5992.

### CODE REVIEW

File: libsolidity/analysis/ReferencesResolver.cpp [hunk 1]
```diff
@@ -298,11 +298,13 @@ bool ReferencesResolver::visit(InlineAssembly const& _inlineAssembly)
...
 			declarations = m_resolver.nameFromCurrentScope(realName);
 		}
-		if (declarations.size() != 1)
+		if (declarations.size() > 1)
 		{
 			declarationError(_identifier.location, "Multiple matching identifiers. Resolving overloaded identifiers is not supported.");
 			return size_t(-1);
 		}
+		else if (declarations.size() == 0)
+			return size_t(-1);
```
Comment (Marenz) [inline libsolidity/analysis/ReferencesResolver.cpp:307]: Don't know what the policy says, but I'd add a blank line after this

**Review by Marenz — DISMISSED** [review]

File: Changelog.md [hunk 2]
```diff
@@ -12,6 +12,7 @@ Bugfixes:
  * ABIEncoderV2: Fix internal error related to bare delegatecall.
  * ABIEncoderV2: Fix internal error related to ecrecover.
  * ABIEncoderV2: Fix internal error related to mappings as library parameters.
+ * Inline Assembly: Proper error message when ``jump`` refers to a missing jump label.
```
Comment (chriseth) [inline Changelog.md]: Is it just for jumps and not for identifiers in general?

Comment (axic) [reply Changelog.md]: Hm, perhaps it is, let me check.

Comment (axic) [reply Changelog.md]: Yes, it is for any identifier:
```
function f() {
  assembly {
     x := 1
  }
}
```

Comment (axic) [reply Changelog.md]: Updated.

**Comment by axic** [comment]
@chriseth updated

File: libsolidity/analysis/ReferencesResolver.cpp (hunk 1 above)
Comment (axic) [reply libsolidity/analysis/ReferencesResolver.cpp:307]: The entire function has no blanks. I'd prefer to have some, but perhaps I'll just leave it as is in this PR.

**Review by chriseth — APPROVED** [review]

**Pull request merged**